test/parse/test_entity_declaration.rb

```diff
@@ -23,6 +23,239 @@ def parse(internal_subset)
     end
 
     public
+
+    class TestGeneralEntityDeclaration < self
+      # https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-GEDecl
+      class TestEntityDefinition < self
+        # https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-EntityDef
+        class TestEntityValue < self
+          def test_no_quote
+            # https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-EntityValue
+            exception = assert_raise(REXML::ParseException) do
+              REXML::Document.new('<!DOCTYPE root [<!ENTITY valid-name invalid-entity-value > ]>')
```

Comment:
Could you remove needless spaces?

```suggestion
              REXML::Document.new('<!DOCTYPE root [<!ENTITY valid-name invalid-entity-value>]>')
```